src/it/resources/com/google/checkstyle/test/chapter5naming/rule523methodnames/InputMethodName.java

```diff
@@ -22,6 +24,24 @@ void f() {} // violation 'Method name 'f' must match pattern'
 
   void fO() {} // violation 'Method name 'fO' must match pattern'
 
+  @Test
+  void testing_foo() {}
+
+  @Test
+  void testing_Foo() {}
```

Comment:
I think there should be violation for this method, after underscore the first character should be lowercase letter. @rdiachenko what do you think about this?

Reply:
but documentation does not explicitly mentions anything about this so I'm not sure if we should give violation for this or not

[5.2.3 Method names](https://google.github.io/styleguide/javaguide.html#s5.2.3-method-names)

Reply:
It should be a violation here. The doc mentions this in the following sentence:

> with each component written in `lowerCamelCase`

where `lowerCamelCase` is example of correct component naming. If the first character after underscore were allowed, they would use this wording instead:

> with each component written in `UpperCamelCase`

Reply:
I see. So there's a defect from our side

Reply:
from coverage table:[5.2.3 Method names](https://checkstyle.org/styleguides/google-java-style-20180523/javaguide.html#s5.2.3-method-names) 	[MethodName](https://checkstyle.org/checks/naming/methodname.html#MethodName)
No ability to distinguish verb at the beginning of name, no way distinguish Test method from others so "_" is allowed in names

I do not understand why this is not a violation:
```
@Test
  void testing_Foo() {}
```

to archive allowance of `_` in test methods we should make Xpath suppression that check that method has `Test` annotation (just text comparison, as there bunch of test libraries in this world, they did not mean only junit).
In addition to this suppression instance we need to add checking for message that contains `_` in message.

in this case  we can remove this `no way distinguish Test method from others so "_" is allowed in names` from comments.

------------

>testing_Foo

from abbreviation perspective this is correct,  camel case is checked by  AbbreviationAsWordInName.
`_` is separator of words. In style guide it is covered by `Divide this result into words, splitting on spaces and any remaining punctuation (typically hyphens). `

Reply:
@Zopsss , please create separate issue on this to clarify it separately.
we can merge this PR and issue will point to merged code and question correctness of violations or missed violations.

Reply:
> @Zopsss, please explain why there is no violation.

defect in our regex:

https://github.com/checkstyle/checkstyle/blob/9da5170cdd0d9e82803c83eb2cc23a0c08d370ee/src/main/resources/google_checks.xml#L353-L357

We only check for starting 2 characters, after that `\w` comes which allows for underscore, digits and alphabets. Then we don't validate anything. That's why there's no violation

Reply:
> @Zopsss , please create separate issue on this to clarify it separately.

done https://github.com/checkstyle/checkstyle/issues/15425

Reply:
> @Zopsss, please explain why there is no violation.

as you explained below in your comment, it is a defect our Check. Once it is fixed, we will have violation for this method

Reply:
>to archive allowance of _ in test methods we should make Xpath suppression that check that method has Test annotation (just text comparison, as there bunch of test libraries in this world, they did not mean only junit).

Let's create issue on this. It is still a gap.

Reply:
https://github.com/checkstyle/checkstyle/commit/0eadd97b8af5b79e7c49f82958fbb3d863bfdf61 is where `_` is explicitly disappear and  becomes covered by `\w`.

Reply:
postponed until https://github.com/checkstyle/checkstyle/issues/15429